core/matchdata/byteoffset_spec.rb

```diff
@@ -0,0 +1,65 @@
+require_relative '../../spec_helper'
+
+describe "MatchData#byteoffset" do
+  ruby_version_is "3.2" do
+    it "returns beginning and ending byte-based offset of n-th match" do
+      m = /(.)(.)(\d+)(\d)/.match("THX1138.")
+
+      m.byteoffset(0).should == [1, 7]
+      m.byteoffset(2).should == [2, 3]
+      m.byteoffset(4).should == [6, 7]
+    end
+
+    it "accepts String as an reference to named capture" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      m.byteoffset("f").should == [0, 3]
+      m.byteoffset("b").should == [3, 6]
+    end
+
+    it "accepts Symbol as an reference to named caputure" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      m.byteoffset(:f).should == [0, 3]
+      m.byteoffset(:b).should == [3, 6]
+    end
+
+    it "returns array of two nils if match value is nil" do
+      m = /(?<x>q..)?/.match("foobarbaz")
+      m.byteoffset("x").should == [nil, nil]
+    end
+
+    it "returns correct beginning and ending byte-based offset for multi-byte strings" do
+      m = /\A\u3042(.)(.)?(.)\z/.match("\u3042\u3043\u3044")
+
+      m.byteoffset(1).should == [3, 6]
+      m.byteoffset(3).should == [6, 9]
+    end
+
+    it "returns array of two nils for wrong argument for multi-byte string" do
+      m = /\A\u3042(.)(.)?(.)\z/.match("\u3042\u3043\u3044")
+
+      m.byteoffset(2).should == [nil, nil]
+    end
+
+    it "raises IndexError if can't find the group" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      -> {
+        m.byteoffset("y")
+      }.should raise_error(IndexError, "undefined group name reference: y")
+
+      -> {
+        m.byteoffset(:y)
+      }.should raise_error(IndexError, "undefined group name reference: y")
+    end
+
+    it "raises IndexError if index is out of matches" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      -> {
+        m.byteoffset(3)
+      }.should raise_error(IndexError, "index 3 out of matches")
+    end
+  end
+end
```

Comment:
**suggestion:** Also some specs on argument type coercion would be useful. I mean cases when not Integer and not String/Symbol is passed (e.g. Float, Rational, or arbitrary Object.new or []).